### PR TITLE
fix(a11y): Add meaningful alt text to custom logo

### PR DESCRIPTION
**Pull request by tille**
- Missing alt attribute doesn't compliant a11y standards.

### CODE REVIEW

File: src/lib/PreviewUI.js [hunk 1]
```diff
@@ -48,6 +48,9 @@ class PreviewUI {
     /** @property {HTMLElement} - Preview container element which houses the sidebar and content */
     previewContainer;
 
+    /** @property {HTMLElement} - Preview container element which houses the custom-logo */
+    customLogoEl;
```
Comment (ConradJChan) [inline src/lib/PreviewUI.js:52]: It doesn't seem like `customLogoEl` is referenced anywhere else, could the `alt` attribute be set as part of the `setupHeader` logic without introducing this class level property?

Comment (tille) [reply src/lib/PreviewUI.js:52]: the problem are these lines:

```
if (options.header !== 'none') {
     this.setupHeader(options.header, options.logoUrl);
```

if there's no `options.header` then `setupHeader` is not call,  so `customLogoEl` exists but it is hidden because the class `CLASS_HIDDEN`

however even if its hidden is an error for a11y, so that's why you need to be sure to always set the `alt` attribute

File: src/lib/PreviewUI.js [hunk 2]
```diff
@@ -385,9 +392,8 @@ class PreviewUI {
             const defaultLogoEl = headerEl.querySelector(SELECTOR_BOX_PREVIEW_LOGO_DEFAULT);
             defaultLogoEl.classList.add(CLASS_HIDDEN);
 
-            const customLogoEl = headerEl.querySelector(SELECTOR_BOX_PREVIEW_LOGO_CUSTOM);
-            customLogoEl.src = logoUrl;
-            customLogoEl.classList.remove(CLASS_HIDDEN);
+            this.customLogoEl.src = logoUrl;
```
Comment (jstoffan) [inline src/lib/PreviewUI.js:395]: I don't think this covers all our use cases, since we have both a "default" element and a "custom" logo element. I suggest we rearrange `setupHeader` to apply the `alt` attribute to both.

File: src/lib/PreviewUI.js [hunk 3]
```diff
@@ -112,6 +115,10 @@ class PreviewUI {
         this.container = this.container.querySelector(SELECTOR_BOX_PREVIEW_CONTAINER);
         this.previewContainer = this.container.querySelector(SELECTOR_BOX_PREVIEW);
         this.contentContainer = this.container.querySelector(SELECTOR_BOX_PREVIEW_CONTENT);
+        this.customLogoEl = this.container.querySelector(SELECTOR_BOX_PREVIEW_LOGO_CUSTOM);
```
Comment (jstoffan) [inline src/lib/PreviewUI.js:118]: We should try to avoid adding instance properties when possible. It seems like we should be able to handle all of this within `setupHeader`.

**Review by jstoffan — COMMENTED** [review]
Thanks for the submission, @tille. I think we still need to iterate on the solution a bit.

File: src/lib/PreviewUI.js (hunk 1 above)
Comment (jstoffan) [reply src/lib/PreviewUI.js:52]: @tille, can you please clarify how this affects the end user? The `CLASS_HIDDEN` class, `bp-is-hidden`, sets `display: none` on the element to which it's applied, which essentially removes it from the DOM as it is not accessible to the user.

File: src/lib/PreviewUI.js [hunk 4]
```diff
@@ -369,11 +369,14 @@ class PreviewUI {
      */
     setupHeader(headerTheme, logoUrl) {
         const headerContainerEl = this.container.querySelector(SELECTOR_BOX_PREVIEW_HEADER_CONTAINER);
-        headerContainerEl.classList.remove(CLASS_HIDDEN);
-
+        const customLogoEl = this.container.querySelector(SELECTOR_BOX_PREVIEW_LOGO_CUSTOM);
```
Comment (jstoffan) [inline src/lib/PreviewUI.js:372]: Can we continue using `headerEl.querySelector` to retrieve this element?

File: src/lib/PreviewUI.js [hunk 5]
```diff
@@ -369,11 +369,14 @@ class PreviewUI {
...
         const headerEl = headerContainerEl.firstElementChild;
+        const defaultLogoEl = headerEl.querySelector(SELECTOR_BOX_PREVIEW_LOGO_DEFAULT);
+
+        headerContainerEl.classList.remove(CLASS_HIDDEN);
```
Comment (jstoffan) [inline src/lib/PreviewUI.js:376]: Can we restore this line to its prior position? It doesn't seem related to the other changes.

File: src/lib/PreviewUI.js [hunk 6]
```diff
@@ -369,11 +369,14 @@ class PreviewUI {
...
         headerEl.className = `${CLASS_BOX_PREVIEW_HEADER} ${CLASS_BOX_PREVIEW_BASE_HEADER}`;
         this.previewContainer.classList.add(CLASS_BOX_PREVIEW_HAS_HEADER);
+        defaultLogoEl.setAttribute('aria-label', __('logo'));
```
Comment (jstoffan) [inline src/lib/PreviewUI.js:379]: It looks like the attribute we're using is inconsistent. Should these both use `alt` or `aria-label`?

Comment (tille) [reply src/lib/PreviewUI.js:379]: for `defaultLogoEl` it has to be `aria-label` because its NOT an `img` element

Comment (tille) [reply src/lib/PreviewUI.js:380]: but is mandatory for ALL the `img` elements to have an `alt` attribute

Comment (jstoffan) [reply src/lib/PreviewUI.js:380]: Good catch. Should we add `role="img"` to the SVG, while we're at it?

**Review by jstoffan — APPROVED** [review]
LGTM, thanks for updating!

File: src/lib/PreviewUI.js (hunk 6 above)
Comment (tille) [reply src/lib/PreviewUI.js:380]: not because ScreenReader would never reach the svg, they will focus the container element that's `defaultLogoEl`

Comment (tille) [reply src/lib/PreviewUI.js:380]: but we can add `role=img` to `<span class="bp-default-logo">`

**Review by jstoffan — APPROVED** [review]
